publication.go

```diff
@@ -242,6 +242,7 @@ type LocalTrackPublication struct {
 	simulcastTracks map[livekit.VideoQuality]*LocalSampleTrack
 	onRttUpdate     func(uint32)
 	opts            TrackPublicationOptions
+	onMuteChanged   func(*LocalTrackPublication, bool)
```

Comment:
Can we make this `OnMuteChanged` (upper case `O`) as it is set directly from the outside?

Reply:
This will cause the callback can be accessed by the user, so keep it lowercase to be access only by internal code as same as other callbacks like onRttUpdate